docs: add integration engineering charter (#400)
Autohive-AI/integrations-autohive: integration-engineering-charter.md

# Integration Engineering Charter

## Preamble

> Autohive integrations connect our users' workflows to third-party systems they rely on. A poor integration does not only produce bad code; it can break customer workflows, mishandle data, request excessive permissions or create long-term maintenance burden for the team.

Reviewer: I'm not concerned about excessive permissions being asked. If we are not covering scopes that aren't in use is a different story. We should make autonamous integrations. But ensure that all scopes are used if mentioned 

 
```suggestion
Autohive integrations connect our users' workflows to third-party systems they rely on. A poor integration does not only produce bad code; it can break customer workflows, mishandle data, or create long-term maintenance burden for the team.
```

Author: But that's what requesting excessive permissions is: We ask for scopes that are not used in the integration.

Reviewer: I'd word it differently in that case. I don't want people to get the wrong picture

Reviewer: I've added a new suggestion below

Author: Cool, thats works imho - added it.

> Autohive integrations connect our users' workflows to third-party systems they rely on. A poor integration does not only produce bad code; it can break customer workflows, mishandle data, request excessive permissions or create long-term maintenance burden for the team.

Reviewer: ```suggestion
Autohive integrations connect our users' workflows to third-party systems they rely on. A poor integration does not only produce bad code; it can break customer workflows, mishandle data, requesting permissions/scopes not used in the integration or create long-term maintenance burden for the team.
```

> AI-assisted development is welcome, but authorship cannot be delegated to AI. The person opening the PR is responsible for understanding the provider API, the Autohive SDK, the implementation choices, the tests, the security implications, and the user experience.

Reviewer: ```suggestion
AI-assisted development is a requirement, but authorship cannot be delegated to AI. The person opening the PR is responsible for understanding the provider API, the Autohive SDK, the implementation choices, the tests, the security implications, and the user experience.
```

### 2. We build for user workflows, not endpoint coverage

A good integration is not a mechanical mirror of a 3rd-party API. It exposes actions that make sense for real Autohive workflows.

We intentionally choose what to include, what to leave out, and how to present actions so users can succeed without needing to understand the provider API.

### 3. We own AI-generated code

AI can help draft, explore, refactor, and test. It cannot fully own correctness.

If AI generated part of an integration, the author is responsible for reviewing it, simplifying it, removing hallucinated behavior, checking it against documentation, and making sure they can explain it.

Code that the author cannot explain is not ready for review.

### 4. We write maintainable Python

Integrations should be idiomatic, readable Python. The code should use simple data structures, clear control flow, accurate type hints where useful, and appropriate async patterns.

We avoid unnecessary abstractions, complicated cleverness, dead code, broad exception swallowing, hidden side effects, and dependencies that are not clearly justified.

### 5. We treat auth and permissions as security decisions

> Scopes, permissions, credentials, files, and user data require care. Integrations should request the minimum access needed for the implemented actions.

Reviewer: ```suggestion
Scopes, permissions, credentials, files, and user data require care. Integrations should request the correct access needed for the implemented actions.
```

> Secrets must never be committed. Test data must be safe. Provider responses should not leak credentials or sensitive data through logs, errors, or outputs.

Reviewer: ```suggestion
Secrets must never be committed in this repo. Test data must be safe. Provider responses should not leak credentials or sensitive data through logs, errors, or outputs.
```

### 6. We verify behavior, not just syntax

Tests should prove that actions behave correctly, handle errors, and match real provider response shapes.

Mocked tests are necessary, but they are not enough for integrations that depend on third-party APIs. Where credentials and safe test data are available, integration tests should exercise real API behavior. If real API tests are not included, the PR should explain why and describe how the real API behavior was verified.

### 7. We document what future maintainers need to know

A good README explains what the integration does, how authentication works, where the official API docs are, what actions exist, and any important provider limitations.

Documentation should describe the integration as shipped, not as imagined future work.

### 8. We self-review before asking others to review

> Opening a PR is a request for another person's time and judgment. Authors should review their own diff first, remove low-quality artifacts, run local validation, check tests, and ensure the PR is focused.

Reviewer: ```suggestion
Opening a PR is a request for another person's time and judgment. Authors should review their own diff first, run the review skill, remove low-quality artifacts, run local validation, check tests, and ensure the PR is focused.
```
